src/contrib/serializers/Akka.Serialization.Hyperion/HyperionSerializer.cs

```diff
@@ -87,5 +110,53 @@ public override object FromBinary(byte[] bytes, Type type)
                 return res;
             }
         }
+
+        private IKnownTypesProvider CreateKnownTypesProvider(ExtendedActorSystem system, Type type)
+        {
+            var ctors = type.GetConstructors();
+            var ctor = ctors.FirstOrDefault(c =>
+            {
+                var parameters = c.GetParameters();
+                return parameters.Length == 1 && (parameters[0].ParameterType == typeof(ActorSystem)
+                    || parameters[0].ParameterType == typeof(ExtendedActorSystem));
+            });
+
+            return ctor == null
+                ? (IKnownTypesProvider) Activator.CreateInstance(type)
+                : (IKnownTypesProvider) ctor.Invoke(new object[] {system});
+        }
+    }
+
+    public sealed class HyperionSerializerSettings
+    {
+        public static readonly HyperionSerializerSettings Default = new HyperionSerializerSettings(
+            preserveObjectReferences: true,
+            versionTolerance: true,
+            knownTypesProvider: typeof(NoKnownTypes));
+
+        public static HyperionSerializerSettings Create(Config config)
+        {
+            if (config == null) throw new ArgumentNullException(nameof(config), "HyperionSerializerSettings require a config, default path: `akka.serializers.hyperion`");
+
+            var type = Type.GetType(config.GetString("known-types-provider"), true);
```

Comment:
Need some specs to verify that all of these settings can be loaded by the Hyperion serializer from a valid HOCON file.